src/ring_theory/algebra_operations.lean

```diff
@@ -173,12 +173,13 @@ begin
     apply mul_subset_mul }
 end
 
-instance span.is_semiring_hom : is_semiring_hom (submodule.span R : set A → submodule R A) :=
-{ map_zero := span_empty,
-  map_one := show _ = map _ ⊤,
+def span.ring_hom : set A →+* submodule R A :=
```

Comment:
Please add a docstring.

Reply:
I don't even know what this is doing... :-) I guess I can work it out.

Reply:
`/-- the span is a ring homomorphism -/` should be good enough.

Reply:
Actually, if someone else could do this, that would be great. I don't see the point of this ring structure, so I'm probably the wrong person to write the doc-string about it.

Reply:
```suggestion
/-- `span` is a semiring homomorphism (recall multiplication is pointwise multiplication of subsets on either side). -/
def span.ring_hom : set A →+* submodule R A :=
```